src/socket/mod.rs

```diff
@@ -32,19 +33,14 @@ pub use self::tcp::TcpSocket;
 pub use self::set::{Set as SocketSet, Item as SocketSetItem, Handle as SocketHandle};
 pub use self::set::{Iter as SocketSetIter, IterMut as SocketSetIterMut};
 
+pub use self::socket_ref::SocketRef;
+pub(crate) use self::socket_ref::SocketSession;
+
 /// A network socket.
 ///
 /// This enumeration abstracts the various types of sockets based on the IP protocol.
 /// To downcast a `Socket` value down to a concrete socket, use
-/// the [AsSocket](trait.AsSocket.html) trait, and call e.g. `socket.as_socket::<UdpSocket<_>>()`.
-///
-/// The `process` and `dispatch` functions are fundamentally asymmetric and thus differ in
-/// their use of the [trait PacketRepr](trait.PacketRepr.html). When `process` is called,
-/// the packet length is already known and no allocation is required; on the other hand,
-/// `process` would have to downcast a `&PacketRepr` to e.g. an `&UdpRepr` through `Any`,
-/// which is rather inelegant. Conversely, when `dispatch` is called, the packet length is
-/// not yet known and the packet storage has to be allocated; but the `&PacketRepr` is sufficient
-/// since the lower layers treat the packet as an opaque octet sequence.
+/// the [FromSocket](trait.FromSocket.html) trait, and call e.g. `UdpSocket::from_socket(socket)`.
```

Comment:
Worth mentioning SocketSet here and that they're probably looking for `socket_set.get::<TcpSocket>(tcp_handle)` instead